### ISSUE TITLE
Active911 Website feature of responding MCEMS members

### ISSUE THREAD

**Issue by EliRuss** [MCEMS/dispatch#13]
Campus Safety is requesting a feature on the dispatch site that allows the Dispatcher to see which members are responding to the call. Can we make this happen over Winter Break?
@jmankhan @benburwell

**Comment by benburwell** [comment]
@EliRuss, just to clarify, this is a list of members responding to each call that is dispatched, correct? I couldn't tell from Brian's email whether he meant that or if he meant just a list of the current status of each member, which there isn't really a way to access (Active911's data model is based around the concept of "alerts").

**Comment by EliRuss** [comment]
The request is not clear, but I believe they want a way for the Dispatcher to be able to see when a member(s) is going to be at that call, when they on their way, possibly when they have arrived. Otherwise, they have no concrete way of knowing if the call will be handled sufficiently without intervention by, for instance, a Line Officer to track it for them. 

They have also expressed a lot of interest in providing times of dispatch, arrival, and clearing - though obviously that's a whole other ball game in a sense.

**Comment by benburwell** [comment]
Okay, so I think what they're asking for is basically a way to see the status of each call after they've dispatched it. This makes a lot of sense. I'll create a mockup that I'd like to get their approval on before we start developing it.

Also, the reason this is a big feature is mostly due to the Active911 "pull-style" integration, as well as changing the UI to support a list of calls. Once this has been done, adding the times should be trivial.

**Comment by EliRuss** [comment]
Hi @jmankhan, @benburwell- 
Just wanted to check-in and see where we are at with this, thank you

**Comment by jmankhan** [comment]
Hey, sorry for the delay, I got a bit busier than expected with family. I hope to get some progress over this weekend. I'll keep you posted

**Comment by EliRuss** [comment]
Would these functions help us out?
1) PC Webview
http://wiki.active911.com/wiki/index.php/PC_Webview

2) Windows PC App (Beta)
http://blog.active911.com/2015/11/introducing-the-windows-pc-app/

**Comment by benburwell** [comment]
1. The webview could fit the bill -- it looks like it's more based around mapping than seeing which units are responding to a particular call, but my guess is that it might do that too. It might be worth giving this a try for a bit during the trial phase and see if more detail is needed/would be helpful to the dispatchers. If that's the case, we may need to do something like that on dispatch.bergems.org. Last time I looked into the webview, it was unavailable to test with on the free plan so I think we'll need to wait and see once we switch to the paid plan.
2. It looks like the PC app provides more or less the same data as the phone app and is more designed for responders than for a "station." Haven't done much research though. My impression is that the webview would be more suitable than this app though.

@jmankhan, why don't you do a basic implementation of this feature and we can show them the webview and this and see how they'd rather use it. I just published v1.0 of [`active911` on npm](https://www.npmjs.com/package/active911), which should be useful. You can get the refresh token from the [agency tab](https://www.active911.com/node/3#/?tab=agency) on the A911 account. I'll add more detailed info about implementation as well.

**Comment by EliRuss** [comment]
Ok, sounds good. Let me know what you decide. The goal is to have our plan in this regard ready to present to our Advisors by the 3rd week of January.

**Comment by benburwell** [comment]
@jmankhan, I created a ticket for you in the API to implement the backend functionality for this. It's fairly complex, so let me know if you have questions/issues. I'll take care of the frontend.

@EliRuss, here is a description of what I'm planning to build:
1. Dispatchers have the ability to view call info for all alerts they have sent out in the past 24 hours. Information available:
   - Time call was dispatched
   - A unique call identifier that may be used for tracking
   - The location of the call
   - The description of the call
   - The current status of each member who is on the call. "Watch" status will be excluded.
   
   This information will be available in tabular format above the existing Dispatch New Call screen. This information may be delayed by up to 30 seconds.
2. Additional information about each call will be available through a modal pop-up window by clicking on the call row:
   - A timestamped log of the call, containing each status change for responding units
   - The details of the call, i.e. the "Additional Information" they provided when sending the alert
3. If the name of a device is changed in the Active911 administration console, it may take up to a day for the change to be reflected in the dispatch console.

Can you look over this and make sure it meets the requirements of the dispatchers?

**Comment by EliRuss** [comment]
Chief Fidati did not have any objections in his reply, but did indicate they may want to have a login account for each Dispatcher.

**Comment by EliRuss** [comment]
Can this be completed by 1/27, Wednesday 5pm ready to be presented to Campus Safety? @jmankhan @benburwell

**Comment by benburwell** [comment]
@EliRuss, we'll give it our best -- Jalal has told me that he won't be able to finish it by then, so I'm going to try to get it done between tonight and tomorrow night. Victor is working on a piece of this as well.

**Comment by benburwell** [comment]
Resolved in 2171308. If you'd like additional features, tweaks, please open a new issue. Thanks!
